app/products/calls/connection/connection.test.ts

```diff
@@ -0,0 +1,130 @@
+import {WebSocketClient} from './websocket_client';
+
+jest.mock('./websocket_client');
+jest.mock('@mattermost/calls/lib');
+
+describe('newConnection', () => {
+    const mockClient = {
+        getWebSocketUrl: jest.fn(() => 'ws://localhost:8065'),
+        getCallsConfig: jest.fn(() => ({})),
+    };
+
+    const mockRTCStats = {
+        iceStats: {
+            succeeded: [
+                {
+                    id: 'candidatePairA',
+                    timestamp: 45,
+                    priority: 45,
+                    state: 'succeeded',
+                    local: {
+                        candidateType: 'host',
+                        protocol: 'udp',
+                        port: 45000,
+                    },
+                    remote: {
+                        candidateType: 'host',
+                        protocol: 'udp',
+                        port: 8443,
+                    },
+                },
+            ],
+            failed: [],
+        },
+    };
+
+    beforeAll(() => {
+        // eslint-disable-next-line
+        // @ts-ignore
+        global.navigator = {};
+
+        // eslint-disable-next-line
+        // @ts-ignore
+        NetworkManager.getClient = jest.fn(() => mockClient);
+
+        // eslint-disable-next-line
+        // @ts-ignore
+        RTCPeer.mockImplementation(() => {
+            return {
+                getStats: jest.fn(),
+                on: jest.fn(),
+            };
+        });
+
+        // eslint-disable-next-line
+        // @ts-ignore
+        RTCMonitor.mockImplementation(() => {
+            return {
+                on: jest.fn(),
+            };
+        });
+
+        Platform.OS = 'web';
+
+        InCallManager.start = jest.fn();
+        InCallManager.stopProximitySensor = jest.fn();
+    });
+
+    afterAll(() => {
+        // eslint-disable-next-line
+        // @ts-ignore
+        delete global.navigator;
+
+        jest.resetAllMocks();
+    });
+
+    it('collectICEStats', (done) => {
+        const wsSend = jest.fn();
+        const joinHandler = jest.fn(async (event: string, cb: () => void) => {
+            if (event === 'join') {
+                await cb();
+                expect(parseRTCStats).toHaveBeenCalled();
+                expect(wsSend).toHaveBeenCalledWith('metric', {
+                    metric_name: 'client_ice_candidate_pair',
+                    data: JSON.stringify({
+                        state: 'succeeded',
+                        local: {
+                            type: 'host',
+                            protocol: 'udp',
+                        },
+                        remote: {
+                            type: 'host',
+                            protocol: 'udp',
+                        },
+                    }),
+                },
+                );
+                done();
+            }
+        });
+
+        // eslint-disable-next-line
+        // @ts-ignore
+        WebSocketClient.mockImplementation(() => {
```

Comment:
Typescript and mocks are a bit painful. This is to avoid this error:

```
error TS2339: Property 'mockImplementation' does not exist on type 'typeof WebSocketClient'.
```

I tried a couple of things (e.g., using `jest.mocked`) but it didn't work well since this is a class and the object is initialized deep inside `newConnection`. I am disabling the type checks because it's becoming a time sink to fix properly. But please let me know if you have any suggestions.

Reply:
No suggestion from me. Thank you for the solution!

Reply:
I had to do the same in other tests